src/modules/ExactTracking.jl

```diff
@@ -106,33 +106,35 @@ properties, though I've not seen a proof of that claim.
 @makekernel fastgtpsa=true function multipole_kick!(i, coords::Coords, ms, knl, ksl, excluding)
   v = coords.v
   alive = ifelse(coords.state[i]==State.Alive, 1, 0) 
-  bx, by = normalized_field!(ms, knl, ksl, v[i,XI], v[i,YI], excluding)
+  bx, by = normalized_field(ms, knl, ksl, v[i,XI], v[i,YI], excluding)
   v[i,PXI] -= by * alive
   v[i,PYI] += bx * alive
 end # function multipole_kick!()
 
 
-@inline function normalized_field!(ms, knl, ksl, x, y, excluding)
+@inline function normalized_field(ms, knl, ksl, x, y, excluding)
   """
   Returns (bx, by), the transverse components of the magnetic field divided
   by the reference rigidty.
   """
-  jm = length(ms)
-  m  = ms[jm]
-  add = (m != excluding && m > 0)
-  by = knl[jm] * add
-  bx = ksl[jm] * add
-  jm -= 1
-  while 2 <= m
-    m -= 1
-    t  = (by * x - bx * y) / m
-    bx = (by * y + bx * x) / m
-    by = t
-    add = (0 < jm && m == ms[jm]) && (m != excluding) # branchless
-    idx = max(1, jm) # branchless trickery
-    by += knl[idx] * add
-    bx += ksl[idx] * add
-    jm -= add
+  @FastGTPSA begin @inbounds begin
+    jm = length(ms)
+    m  = ms[jm]
+    add = (m != excluding && m > 0)
+    by = knl[jm] * add
+    bx = ksl[jm] * add
+    jm -= 1
+    while 2 <= m
+      m -= 1
+      t  = (by * x - bx * y) / m
+      bx = (by * y + bx * x) / m
+      by = t
+      add = (0 < jm && m == ms[jm]) && (m != excluding) # branchless
+      idx = max(1, jm) # branchless trickery
+      by += knl[idx] * add
+      bx += ksl[idx] * add
+      jm -= add
+    end end
```

Comment:
```suggestion
  @FastGTPSA begin
    jm = length(ms)
    m  = ms[jm]
    add = (m != excluding && m > 0)
    by = knl[jm] * add
    bx = ksl[jm] * add
    jm -= 1
    while 2 <= m
      m -= 1
      t  = (by * x - bx * y) / m
      bx = (by * y + bx * x) / m
      by = t
      add = (0 < jm && m == ms[jm]) && (m != excluding) # branchless
      idx = max(1, jm) # branchless trickery
      by += knl[idx] * add
      bx += ksl[idx] * add
      jm -= add
    end
```

Note that `@makekernel` automatically wraps everything in an `@inbounds`, which is then propagated into calling functions. So here we do not need a `@inbounds`

Reply:
Doesn't that apply to FastGTPSA too then?